static/bootstrap-utils.ts

```diff
@@ -0,0 +1,471 @@
+// Copyright (c) 2025, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/**
+ * TEMPORARY COMPATIBILITY LAYER
+ *
+ * This module provides utilities to help transition from Bootstrap 4's jQuery-based API
+ * to Bootstrap 5's vanilla JavaScript API. This is intended as a temporary solution
+ * during the migration from Bootstrap 4 to 5 and should be removed once the migration
+ * is complete.
+ *
+ * The goal is to minimize changes throughout the codebase by centralizing the Bootstrap
+ * API changes in this file, while still allowing for gradual migration to direct API calls.
+ *
+ * @deprecated This module should be removed after the Bootstrap 5 migration is complete.
+ */
+
+import $ from 'jquery';
+
+import 'bootstrap';
+import {Collapse, Dropdown, Modal, Popover, Tab, Toast, Tooltip} from 'bootstrap';
+
+export class BootstrapUtils {
```

Comment:
If there are only static methods on this class I much rather prefer just exporting functions from top level instead.

Reply:
Will fix :)